src/test/functional/pages/PimPage.ts

```diff
@@ -1,6 +1,63 @@
-import { BasePage } from "./BasePage";
+import { Locator, Page } from 'playwright';
 
+import { newEmployeeData } from '../data';
+
+import { BasePage } from './BasePage';
+import { PersonalDetailsPage } from './PersonalDetailsPage';
+
+const randomNumber = (Math.random() * 1000).toString().substring(0, 3);
 export class PimPage extends BasePage {
+  readonly page: Page;
+  readonly pimPageButton: Locator;
+  readonly addButton: Locator;
+  readonly nameInput: Locator;
+  readonly middleNameInput: Locator;
+  readonly lastNameInput: Locator;
+  readonly employeeId: Locator;
+  readonly createLoginDetailsCheckbox: Locator;
+  readonly username: Locator;
+  readonly password: Locator;
+  readonly passwordConfirmed: Locator;
+  readonly saveButton: Locator;
+  readonly cancelButton: Locator;
+  readonly personalDetailsPage: Page;
+
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.pimPageButton = page.getByRole('link', { name: 'PIM' });
```

Comment:
Isn't is redundant since you have openSubPage (or something like that) in BasePage?